purchase_stock_price_variance/models/stock_picking.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class Stockpick(models.Model):
+    _inherit = "stock.picking"
+
+    bypass_price_variance_check = fields.Boolean(
+        copy=False,
+        tracking=True,
+        help="If enabled, no error is raised for price variance between "
+        "the product's standard price and purchase receipt unit price.",
+    )
+
+    enable_price_variance_error = fields.Boolean(
+        compute="_compute_enable_price_variance_error",
+        store=True,
+    )
+
+    @api.depends("company_id", "company_id.enable_price_variance_error")
+    def _compute_enable_price_variance_error(self):
+        for picking in self:
+            picking.enable_price_variance_error = (
+                picking.company_id.enable_price_variance_error
+                if picking.company_id
+                else False
+            )
+
+    def write(self, vals):
+        if "bypass_price_variance_check" in vals:
+            if not self.env.user.has_group(
+                "purchase_stock_price_variance.group_bypass_price_variance_check"
+            ):
+                raise UserError(
+                    _(
+                        "You do not have permission to modify the "
+                        "'Bypass Price Variance Check' field. "
+                        "Please contact an administrator or a user "
+                        "with the appropriate permissions."
+                    )
+                )
+        return super().write(vals)
+
+    def _action_done(self):
+        global_price_variance_threshold_percent = (
+            self.env.company.price_variance_threshold_percent
+        )
+        global_price_variance_threshold_amount = (
+            self.env.company.price_variance_threshold_amount
+        )
```

Comment:
```suggestion
        company = self.env.company
        global_threshold_percent = company.price_variance_threshold_percent
        global_threshold_amount = company.price_variance_threshold_amount
```
Not a fan of log variable names in methods. ;)